Sources/DynamicNotchKit/DynamicNotchInfo.swift

```diff
@@ -7,47 +7,76 @@
 
 import SwiftUI
 
-internal final class DynamicNotchInfoPublisher: ObservableObject {
-    @Published var icon: Image?
+internal final class DynamicNotchInfoPublisher<IconView>: ObservableObject where IconView: View {
+    @Published var iconView: IconView?
     @Published var iconColor: Color
     @Published var title: String
     @Published var description: String?
 
-    init(icon: Image?, iconColor: Color, title: String, description: String? = nil) {
-        self.icon = icon
+    init(icon: Image?, iconColor: Color, title: String, description: String? = nil) where IconView == Image {
+        self.iconView = icon ?? nil
         self.iconColor = iconColor
         self.title = title
         self.description = description
     }
 
+    init(iconView: IconView?, title: String, description: String? = nil) {
+        self.iconView = iconView
+        self.iconColor = .clear
+        self.title = title
+        self.description = description
+    }
+
     @MainActor
-    func publish(icon: Image?, iconColor: Color, title: String, description: String?) {
-        self.icon = icon
+    func publish(icon: Image?, iconColor: Color, title: String, description: String?) where IconView == Image {
+        self.iconView = icon
         self.iconColor = iconColor
         self.title = title
         self.description = description
     }
+
+    @MainActor
+    func publish(icon: IconView?, title: String, description: String?) {
+        self.iconView = icon
+        self.title = title
+        self.description = description
+    }
 }
 
-public class DynamicNotchInfo {
+public class DynamicNotchInfo<IconView> where IconView: View {
     private var internalDynamicNotch: DynamicNotch<InfoView>
-    private let publisher: DynamicNotchInfoPublisher
+    private let publisher: DynamicNotchInfoPublisher<IconView>
 
-    public init(contentID: UUID = .init(), icon: Image! = nil, title: String, description: String? = nil, iconColor: Color = .white, style: DynamicNotch<InfoView>.Style = .auto) {
+    public init(contentID: UUID = .init(), icon: Image! = nil, title: String, description: String? = nil, iconColor: Color = .white, style: DynamicNotch<InfoView>.Style = .auto) where IconView == Image {
         let publisher = DynamicNotchInfoPublisher(icon: icon, iconColor: iconColor, title: title, description: description)
         self.publisher = publisher
         internalDynamicNotch = DynamicNotch(contentID: contentID, style: style) {
             InfoView(publisher: publisher)
         }
     }
 
+    public init(contentID: UUID = .init(), iconView: IconView, title: String, description: String? = nil, style: DynamicNotch<InfoView>.Style = .auto) {
```

Comment:
Tried it, seems to work well! I think it would be ideal to have an additional initializer like this too:

```swift
public init(contentID: UUID = .init(), title: String, description: String? = nil, style: DynamicNotch<InfoView>.Style = .auto, iconView: @escaping () -> IconView) {
    let publisher = DynamicNotchInfoPublisher<IconView>(iconView: iconView(), title: title, description: description)
    self.publisher = publisher
    internalDynamicNotch = DynamicNotch(contentID: contentID, style: style) {
        InfoView(publisher: publisher)
    }
}
```

Functionality-wise, it would stay the same, but it would allow the user to specify the view directly inside a function in an escaping closure, which may be preferred in some cases:

```swift
Button("Test") {
    let notch = DynamicNotchInfo(title: "Test") {
        RoundedRectangle(cornerRadius: 5)
            .fill(LinearGradient(colors: [.red, .blue], startPoint: .leading, endPoint: .trailing))
            .aspectRatio(contentMode: .fit)
    }
    notch.show(for: 2)
}
```

But maybe there's a reason you decided against this? I'd love to hear what you think about this :)

Reply:
Oh, that's a good addition, no particular reason why I went for this approach.
I'll fix it up!

Reply:
@MrKai77 pushed a new commit now ;)
https://github.com/MrKai77/DynamicNotchKit/pull/17/commits/4c05b8a040cfa5713f01dfe31ab774350e9d9e22